gemini/tuning/gemini_supervised_fine-tuning_text_classification/gemini_supervised_finetuning_text_classification.ipynb

```diff
@@ -0,0 +1,2724 @@
+        "    <model>\n",
+        "    {train.loc[train[\"label_text\"] == \"business\", \"label_text\"].iloc[10]}\n",
+        "\n",
+        "- EXAMPLE 2:\n",
+        "    <user>\n",
+        "    {train.loc[train[\"label_text\"] == \"entertainment\", \"text\"].iloc[10]}\n",
+        "    <model>\n",
+        "    {train.loc[train[\"label_text\"] == \"entertainment\", \"label_text\"].iloc[10]}\n",
+        "\n",
+        "- EXAMPLE 3:\n",
+        "    <user>\n",
+        "    {train.loc[train[\"label_text\"] == \"politics\", \"text\"].iloc[10]}\n",
+        "    <model>\n",
+        "    {train.loc[train[\"label_text\"] == \"politics\", \"label_text\"].iloc[10]}\n",
+        "\n",
+        "- EXAMPLE 4:\n",
+        "    <user>\n",
+        "    {train.loc[train[\"label_text\"] == \"sport\", \"text\"].iloc[10]}\n",
+        "    <model>\n",
+        "    {train.loc[train[\"label_text\"] == \"sport\", \"label_text\"].iloc[10]}\n",
+        "\n",
+        "- EXAMPLE 4:\n",
+        "    <user>\n",
+        "    {train.loc[train[\"label_text\"] == \"tech\", \"text\"].iloc[10]}\n",
+        "    <model>\n",
+        "    {train.loc[train[\"label_text\"] == \"tech\", \"label_text\"].iloc[10]}\n",
+        "\n",
+        "\"\"\""
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "FMVdJ2EffcuC"
+      },
+      "source": [
+        "For the below evaluations, we'll use the respective functions we have already set up. For in-context learning, we recommend to use the validation set to find the optimal performance and then apply it to the test set, to make sure the metrics remain consistent. In this notebook, we'll directly evaluate on the test dataset, as the validation and prompt engineering part has been already done."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "wF_tas6BfcuC"
+      },
+      "source": [
+        "### 3.1 Gemini 1.0 Pro in-context Evaluation\n",
+        "\n",
+        "#### Zero-Shot Evaluation"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "2460cc454327"
+      },
+      "source": [
+        "First we will compute the predictions using the frozen model with a prompt without examples (i.e. using the `system_prompt_zero_shot` prompt). We also configure the temperature to 0, to make it as grounded as possible,  and max output tokens to 10, as the categories are only one word, we don't need  more than that.\n",
+        "\n",
+        "We are also going to set the safety filters to only block responses which have high severity scores for all four categories."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "OLitVLOFqPQe"
+      },
+      "outputs": [],
+      "source": [
+        "generation_config = {\"max_output_tokens\": 10, \"temperature\": 0}\n",
```

Comment:
Please use the `GenerationConfig` class. I'd also recommend just re-using the same `generation_config` and `safety_settings` throughout the tests if possible.